packages/ui-components/src/components/info-label/info-label.tsx

```diff
@@ -0,0 +1,114 @@
+import { ResizeSensor } from 'css-element-queries';
+import clipboardHelper from '../../utils/clipboard.helper';
+import { COPY_TOOLTIP } from './info-label.types';
+
+@Component({
+	tag: 'kv-info-label',
+	styleUrls: {
+		night: 'info-label.night.scss',
+		light: 'info-label.light.scss'
+	},
+	shadow: true
+})
+export class KvInfoLabel {
+	private descriptionContainer: Element;
+	private tooltipConfig = COPY_TOOLTIP;
+
+	/** Define if the show more button needed to be visible */
+	@State() enableShowMoreButton = false;
+	/** Define if the label content is completely visible */
+	@State() showMore = false;
+	/** Store the current div content height */
+	@State() currentDescriptionHeight: number = null;
+
+	/** (required) Info label title */
+	@Prop({ reflect: true }) labelTitle!: string;
+	/** (optional) Info label description */
+	@Prop({ reflect: true }) description?: string;
+	/** (optional) Info label description height */
+	@Prop({ reflect: true }) descriptionHeight?: number;
+	/** (optional) Info label description collapse text */
+	@Prop({ reflect: true }) descriptionCollapsedText?: string;
+	/** (optional) Info label description opened text */
+	@Prop({ reflect: true }) descriptionOpenedText?: string;
+	/** (optional) Info label copy value */
+	@Prop({ reflect: true }) copyValue?: string;
+
+	/** The Host's element reference */
+	@Element() el: HTMLKvInfoLabelElement;
+
+	get showMoreButtonLabel() {
+		return this.showMore ? this.descriptionOpenedText : this.descriptionCollapsedText;
+	}
+
+	private loadDescriptionHeight() {
+		if (this.descriptionHeight !== undefined) {
+			const descriptionHeight = this.descriptionContainer.clientHeight;
+			this.enableShowMoreButton = descriptionHeight !== undefined && descriptionHeight > this.descriptionHeight;
+		} else {
+			this.enableShowMoreButton = false;
+		}
+		this.showMore = this.enableShowMoreButton ? this.showMore : false;
+		this.currentDescriptionHeight = this.enableShowMoreButton && !this.showMore ? this.descriptionHeight : null;
+	}
+
+	private onShowMoreToggle = () => {
+		this.showMore = !this.showMore;
+
+		if (this.showMore) {
+			const textElementHeight = this.descriptionContainer.clientHeight;
+			this.currentDescriptionHeight = textElementHeight;
+			return;
+		}
+
+		this.currentDescriptionHeight = this.descriptionHeight;
+	};
+
+	private onClickCopyAction = async () => {
+		const tooltip: HTMLKvTooltipElement = this.el.shadowRoot.querySelector('kv-tooltip');
+		const tooltipText = tooltip.shadowRoot.querySelector('#tooltip') as HTMLElement;
+		if (await clipboardHelper.copyTextToClipboard(this.copyValue)) {
+			tooltipText.innerText = this.tooltipConfig.copyTooltipResultLabel;
+		} else {
+			console.error('Copy to clipboard failed');
+		}
+	};
+
+	componentDidRender() {
+		this.descriptionContainer = this.el.shadowRoot.querySelector('.description');
+
+		if (this.descriptionHeight > 0) {
+			new ResizeSensor(this.descriptionContainer, () => {
```

Comment:
we don't need to destruct ResizeSensor when the component unmount?

Reply:
Yes! You're right! I'll do that! Thanks!